packages/smooth_app/lib/bottom_sheet_views/user_preferences_view.dart

```diff
@@ -22,143 +26,232 @@ class UserPreferencesView extends StatelessWidget {
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+            Column(
+              mainAxisSize: MainAxisSize.max,
+              mainAxisAlignment: MainAxisAlignment.end,
+              children: <Widget>[
+                ClipRect(
+                  child: BackdropFilter(
+                    filter: ImageFilter.blur(
+                      sigmaX: 4.0,
+                      sigmaY: 4.0,
+                    ),
+                    child: Container(
+                      color: Colors.black12,
+                      padding: const EdgeInsets.symmetric(
+                          horizontal: 12.0, vertical: 20.0),
+                      child: SmoothMainButton(
+                        text: 'OK',
+                        onPressed: () {
+                          Navigator.pop(context);
+                          if (callback != null) {
+                            callback();
+                          }
+                        },
                       ),
                     ),
                   ),
-                ],
-              ),
-            ],
-          ),
+                ),
+              ],
+            ),
+          ],
         ),
       ),
     );
   }
 
-  Widget _generatePreferenceRow(String variable) {
+  Widget _generatePreferenceRow(
+    final Attribute variable,
+    final double screenWidth,
+    final UserPreferences userPreferences,
+    final UserPreferencesModel userPreferencesModel,
+  ) {
+    final double iconWidth =
+        (screenWidth - _TYPICAL_PADDING_OR_MARGIN * 5) * _PCT_ICON;
+    final double sliderWidth =
+        (screenWidth - _TYPICAL_PADDING_OR_MARGIN * 5) * (1 - _PCT_ICON);
+    final PreferencesValue importance =
+        userPreferencesModel.getPreferencesValue(
+      variable.id,
+      userPreferences,
+    );
     return Container(
-      padding: const EdgeInsets.all(10.0),
-      margin: const EdgeInsets.all(10.0),
+      padding: const EdgeInsets.all(_TYPICAL_PADDING_OR_MARGIN),
+      margin: const EdgeInsets.all(_TYPICAL_PADDING_OR_MARGIN),
+      width: screenWidth,
       decoration: BoxDecoration(
           color: Colors.black.withAlpha(5),
           borderRadius: const BorderRadius.all(Radius.circular(20.0))),
-      child: Column(
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.center,
         children: <Widget>[
-          Row(
-            mainAxisSize: MainAxisSize.max,
-            mainAxisAlignment: MainAxisAlignment.spaceBetween,
-            children: <Widget>[
-              Text(UserPreferencesModel.getVariableName(variable)),
-            ],
+          Container(
+            width: iconWidth,
+            child: SvgPicture.network(
+              variable.iconUrl,
+              alignment: Alignment.centerLeft,
+              width: iconWidth,
+              fit: BoxFit.scaleDown,
```

Comment:
Most of the icons need to be scaled up (I drew them on a 24x24 pixel grid), so we should specify BoxFit.contain

```suggestion
              fit: BoxFit.contain,
```